### DISCUSSION TITLE
umi 3.x下@formily/antd 组件样式不生效

### DISCUSSION THREAD

**Discussion by Slebee** [alibaba/formily#1097] [Q&A 2.x]
用 @formily 封装了一个组件库，umi项目install这个组件库之后，使用封装后的组件出现组件内引用的antd组件样式全部丢失的情况，不会走umi内部antd的按需加载，需要全量引入 import 'antd/dist/antd.css'; 才能正常展示，有什么办法解决么

**Comment by janryWang** [comment]
需要手动配置一下babel-import-plugin

**Reply by Slebee** [reply]
配置了，但是死活不生效，头大，求助

**Reply by studyweb365** [reply]
// 配置babel-plugin-import，让在使用@formily/antd按需加载，保证样式不丢失
  extraBabelPlugins: [
    [
      'babel-plugin-import',
      { libraryName: 'antd', libraryDirectory: 'lib', style: true },
      'antd',
    ],
    [
      'babel-plugin-import',
      { libraryName: '@formily/antd', libraryDirectory: 'lib', style: true },
    ],
  ],

**Reply by studyweb365** [reply]
不过依然会有部分组件样式没有加载过来，比如
![image](https://user-images.githubusercontent.com/69340754/123250060-3b8f9c00-d51c-11eb-8ed5-bb7c107f2894.png)

**Reply by Deja-vuuu** [reply]
这个最后咋解决？ 我也遇到了这种问题。而且我发现用umi初始化项目。安装family会出现版本冲突。目前通过jiang降级npm到6.14.1解决了。
![image](https://user-images.githubusercontent.com/27748682/129712533-fb924b6c-1d83-418b-b729-24fe8b28de98.png)

**Reply by WLyKan** [reply]
> 需要手动配置一下babel-import-plugin

应该是`babel-plugin-import`

**Comment by WLyKan** [comment]
添加了umirc配置
```
extraBabelPlugins: [
    ["import", { libraryName: "antd", style: true }],
  ],
```
我还是没样式，环境是

- "umi": "^3.5.21"
- formily "^2.0.14",
- "antd": "^4.18.9"
- pnpm 6.32.3
- node v14.19.0

**Reply by WLyKan** [reply]
可以正常渲染form样式了，要配置`babel-plugin-import`
```js
  extraBabelPlugins: [
    [
      'babel-plugin-import',
      { libraryName: 'antd', libraryDirectory: 'lib', style: true },
    ],
    [
      'babel-plugin-import',
      { libraryName: '@formily/antd', libraryDirectory: 'lib', style: true },
      'formilyAntd',
    ],
  ],
```

**Reply by WLyKan** [reply]
[文档](https://github.com/alibaba/formily/blob/cf93945d4b/docs/guide/advanced/build.zh-CN.md)里的babel配置会报错，两个独立的plugin 实例， 需要指定一个uniqname

**Reply by WLyKan** [reply]
本来想提交PR，发现在next分支已经修改过了，半年前的。。。那么，什么时候可以更新一个新版本的文档呢，或者，能为文档更新做些什么
